packages/angular_devkit/build_optimizer/src/transforms/wrap-enums.ts

```diff
@@ -0,0 +1,306 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import * as ts from 'typescript';
+import { collectDeepNodes, drilldownNodes } from '../helpers/ast-utils';
+
+
+export function testWrapEnums(content: string) {
+  const regexes = [
+    // tslint:disable:max-line-length
+    /var (\S+) = \{\};\r?\n(\1\.(\S+) = \d+;\r?\n)+\1\[\1\.(\S+)\] = "\4";\r?\n(\1\[\1\.(\S+)\] = "\S+";\r?\n*)+/,
+    /var (\S+);(\/\*@__PURE__\*\/)*\r?\n\(function \(\1\) \{\s+(\1\[\1\["(\S+)"\] = 0\] = "\4";(\s+\1\[\1\["\S+"\] = \d\] = "\S+";)*\r?\n)\}\)\(\1 \|\| \(\1 = \{\}\)\);/,
+  // tslint:enable:max-line-length
+  ];
+
+  return regexes.some((regex) => regex.test(content));
+}
+
+interface EnumData {
+  name: string;
+  hostNode: ts.Node;
+  statements: ts.ExpressionStatement[];
+  dropNodes: ts.Node[];
+}
+
+export function getWrapEnumsTransformer(): ts.TransformerFactory<ts.SourceFile> {
+  return (context: ts.TransformationContext): ts.Transformer<ts.SourceFile> => {
+    const transformer: ts.Transformer<ts.SourceFile> = (sf: ts.SourceFile) => {
+
+      const enums = findEnumDeclarations(sf);
+      const dropNodes = enums.reduce((acc: ts.Node[], curr) => acc.concat(curr.dropNodes), []);
+
+      const visitor: ts.Visitor = (node: ts.Node): ts.Node => {
```

Comment:
I did a PR for that, should be in one of the newer versions https://github.com/Microsoft/TypeScript/pull/17044